### ISSUE TITLE
The command `claudia create ...` does not work if the package name is `@company/xyz`

### ISSUE THREAD

**Issue by ncochard** [claudiajs/claudia#80]
Hello

I'm trying to use ClaudiaJS in a package named `@company/xyz`. The command `claudia create --region eu-central-1 --api-module index` fails with the following error.

```
Error: ENOENT: no such file or directory, open 'C:\Users\MyUserName\AppData\Local\Temp\3e7c53db-4
43f-438c-bff0-40fbbc715edf\@company\xyz-1.0.0.tgz'
```

(I'll try to provide a pull request in a few days to fix this issue.)

Regards

**Comment by gojko** [comment]
thanks for reporting it. if you want to work on the fix, this is probably the culprit:

https://github.com/claudiajs/claudia/blob/e6f0d885c7cf396263556170079194066bd90bdc/src/tasks/collect-files.js#L41

please also add a test for this (a single test just proving that it can do the package with scoped name): https://github.com/claudiajs/claudia/blob/e6f0d885c7cf396263556170079194066bd90bdc/spec/collect-files-spec.js#L116

**Comment by ncochard** [comment]
Hello

Thanks. The fix is the following, in case you want merge it...

	copyFiles = function (packageConfig) {
		...
		var expectedName = packageConfig.name.split("/").join("-") + '-' + packageConfig.version + '.tgz';
        if (expectedName.charAt(0) === "@") {
        	expectedName = expectedName.substring(1);
		}
		...
	}

I'll send a pull-request as soon as I have finished implementing the corresponding test-case.

Regards

**Comment by gojko** [comment]
maybe the best thing is to extract the `expectedName` calculation into a separate function in util and just write a few unit tests for that, so you don't have to test the whole collectFiles workflow

**Comment by ncochard** [comment]
Hello

I've made progress testing the claudia with a package named `@company/xyz`. You can see my fork at:
[https://github.com/ncochard/claudia](https://github.com/ncochard/claudia).

In the AWS console, I can confirm that the **IAM role**, the **Lambda** and the **API Gateway** were configured. So the process was almost complete before the following error appears.

	BadRequestException: Resource's path part only allow a-zA-Z0-9._- and curly braces at the beginning and the end.

Any pointer would be appreciated.

Regards

**Comment by ncochard** [comment]
Please ignore the last comment. The error `BadRequestException` was unrelated. 
The fork [https://github.com/ncochard/claudia](https://github.com/ncochard/claudia) seems to be working.

**Comment by gojko** [comment]
why did you have to do this?
```
if(typeof apiModule === "object" && apiModule !== null && typeof apiModule.default === "object") {
```

**Comment by ncochard** [comment]
This is to fix another issue which I think is caused by the fact that I'm compiling my module using TypeScript. I can remove this from my pull-request and deal with this separately. I'll send the pull-request when I have found the time to test it properly.

**Comment by gojko** [comment]
this is now available on NPM as version 2.1.3
